### ISSUE TITLE
Large number of matplotlib warnings when running with DEBUG options

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/gammasim-tools#244]
Running some applications (e.g. derrive_mirro_rnda) with -v DEBUG results in a large number of DEBUG output from matplotlib (matplotlib.font_manager and matplotlib.backend_pdf) which makes the output very hard to parse.

I can suppress these messages with
```
logging.getLogger('matplotlib.font_manager').disabled = True
logging.getLogger('matplotlib.backends.backend_pdf').disabled = True
```
and obtain a much more readable and therefore useful output.

I would like to add these two lines? Is there any issue with that?
Suppressing debug messages is not in all cases good, but do we expect that we need those matplotlib debug messages?

**Comment by orelgueta** [comment]
I don't see an issue with that. I agree that we don't need those matplotlib debug messages. The only thing is that we would have to add those lines in every single application/module in which we import the logger and matplotlib. There are quite a few of those places.

In the past we had a separate logger class in gammasim-tools which we imported everywhere, but Raul realised that it's not good practice, so we probably shouldn't go back to that.

Need to look around a bit and see if there is a good solution.

**Comment by orelgueta** [comment]
Taking a very quick look and running some tests, one solution would be to name our logger something like "gammasim-tools-logger" everywhere. This means we will only collect our logging info, excluding any logging info from external packages. Also, when we set the log level, I think it would affect only "our" logger. This is probably not ideal, since perhaps sometimes we will want to look at the log of external packages. However, that might be solved by keeping two log files (streams), one would be the root one that will be from external packages and a gammasim-tools one. 

Implementing this would require a little bit of work and reading what is the best way to do it. Perhaps for now we simply use a named log everywhere and open a separate issue to improve the logger to use files, separate streams, etc. What do you think?
I should point out that this is how we started with the logger and then Raul changed it to the way it is now. I assume he had a good reason, so we might find out in the future that doing it was a mistake...

**Comment by GernotMaier** [comment]
OK - I really don't know enough about this topic to suggest which solution is better. Let me also spent some time in learning a bit about it.

I agree that adding my suggested two lines to each applications is a bit cumbersome (but maybe not: it would only be those applications with plotting involved).

**Comment by orelgueta** [comment]
> I agree that adding my suggested two lines to each applications is a bit cumbersome (but maybe not: it would only be those applications with plotting involved).

Yes, except it would not be only applications, we would have to add it also to modules which use matplotlib. There aren't that many, probably just 3-4, but it's a weird piece of code to add everywhere in the library I think.

I will wait for you to learn a bit about it and we can decide later.

**Comment by GernotMaier** [comment]
Just want to mention that I've struggled with the large number of matplotlib output today again while debugging.

Would it be reasonable to add the two lines suggested at least to the integration test routines? This is what I am using a lot for debugging, and it is very hard to find issues right now.

@orelgueta ?

**Comment by orelgueta** [comment]
Sure, just please mark it with a TODO to look into it (and perhaps keep this issue open).
